lib/python/bind_slice_methods.h

```diff
@@ -89,24 +90,9 @@ auto get_slice_range(T &self, const std::tuple<Dim, const py::slice> &index) {
   auto [dim, py_slice] = index;
   if constexpr (std::is_same_v<T, DataArray> || std::is_same_v<T, Dataset>) {
     try {
-      auto start = py::getattr(py_slice, "start");
-      auto stop = py::getattr(py_slice, "stop");
-      if (!start.is_none() || !stop.is_none()) { // Means default slice : is
-                                                 // treated as index slice
```

Comment:
I think you may have broken this part when you extracted the function?

Reply:
Yes, noticed that, pushed fix now.